src/main/java/org/booking/AccountServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+package org.booking;
+import org.booking.entity.Client;
+import org.booking.entity.Room;
+import org.booking.repository.ClientRepository;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import static org.booking.entity.Status.AVAILABLE;
+import static org.booking.entity.Status.RESERVED;
+
+public class AccountServiceImpl implements AccountService {
+    Scanner scanner = new Scanner(System.in);
+    ClientRepository clientRepository;
+    public AccountServiceImpl(ClientRepository clientRepository) {
+        this.clientRepository = clientRepository;
+    }
+    @Override
+    public Client createClient(Integer clientId, String clientName) {
+        if(clientName.trim().isEmpty()){
+            throw new IllegalArgumentException("Client's name is empty");
+        }
+        return clientRepository.saveClient(new Client(clientId, clientName, new ArrayList<>()));
+    }
+
+    @Override
+    public List<Room> createRooms(int totalRooms) {
+        RoomPriceAndSorting roomPriceAndSorting = new RoomPriceAndSorting(scanner);
+        return roomPriceAndSorting.setPriceForRoomId(totalRooms);
+    }
+    @Override
+    public void paginationRoomForClientRequest(int roomPerPage, List<Room> roomList) {
+        int numberOfRooms = roomList.size();
+        int totalPage = (int) Math.ceil((double) numberOfRooms / roomPerPage);
+        int currentPage = 1;
+        boolean continueNavigation = true;
+
+        while (continueNavigation) {
+            int start = (currentPage - 1) * roomPerPage;
+            int end = Math.min(start + roomPerPage, numberOfRooms);
+
+            System.out.println("\nPage %d of %d".formatted(currentPage, totalPage));
+
+            for (int i = start; i < end; ++i) {
+                System.out.println("""
+                        Room ID: %s,
+                        Status: %s,
+                        Price: %s
+                        """.formatted(
+                        roomList.get(i).getRoomId(),
+                        roomList.get(i).getStatus().name(),
+                        roomList.get(i).getRoomPrice()
+                ));
+            }
+
+            System.out.println("Enter 'n' for next, 'p' for previous");
+            String action = scanner.next();
+            if (currentPage >= totalPage && action.equalsIgnoreCase("n")) {
+                continueNavigation = false;
+            }
+
+            if (action.equalsIgnoreCase("n")) {
+                if (currentPage < totalPage) {
+                    currentPage++;
+                } else {
+                    System.out.printf("You are on the last page: %d", currentPage);
+                }
+            } else if (action.equalsIgnoreCase("p")) {
+                if (currentPage > 1) {
+                    currentPage--;
+                } else {
+                    System.out.printf("You are on the first page: %d", currentPage);
+                }
+            }
+        }
```

Comment:
complex logiv. try to potimize it. or at least, cretae smaller methods and just use them)